### PR TITLE
FedCM docs updates

### PR DESCRIPTION
**Pull request by mihajlija**
Changes proposed in this pull request:

- Restructure the main FedCM doc into overview and developer guide
- Add a link to the developer guide under the existing anchor '#use-api' in the overview doc
- Add auto-reauthn to developer guide
- Update the FedCM changelog doc
- Add the developer guide to /privacy-sandbox
- Remove the mentions of "proposal" status since the API is shipped

### CODE REVIEW

File: site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md [hunk 1]
```diff
@@ -0,0 +1,705 @@
...
+
+{% endAside %}
+
+Once the token is validated by the RP server, the RP may register the user or
+let them sign-in and start a new session.
+
+### Auto-reauthenticate users after the initial consent {: #auto-reauthn }
+
+[FedCM 
+](https://github.com/fedidcg/FedCM/issues/429)[auto-reauthentication](https://github.com/fedidcg/FedCM/issues/429)
```
Comment (agektmr) [inline site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md:565]:
```suggestion
[FedCM auto-reauthentication](https://github.com/fedidcg/FedCM/issues/429) 
```

File: site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md [hunk 2]
```diff
@@ -0,0 +1,705 @@
...
+it is unnecessarily cumbersome after the user has gone through it once: after 
+the user grants permission to allow communication between the RP and the IdP,  
+there's no privacy or security benefit for enforcing another explicit user 
+confirmation for something that they have already previously acknowledged. 
+
+{% Aside %}
+
+Auto-reauthentication is available in [FedCM from Chrome 115](/blog/fedcm-auto-reauthn).
+
+{% endAside %}
```
Comment (agektmr) [inline site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md:583]: I don't think this is needed as we have a update log article.

File: site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md [hunk 3]
```diff
@@ -0,0 +1,705 @@
...
+  <figcaption>A user auto-reauthenticating through FedCM.</figcaption>
+</figure>
+
+#### Enforce mediation with `preventSilentAccess()` {: #prevent-silent-access }
+
+Auto-reauthenticating users immediately after they sign out would not make for a 
+very good user experience. That's why FedCM has a 10-minute quiet period after 
+an auto-reauthn to prevent this behavior. This means that auto-reauthn happens 
+at most once in every 10-minutes unless the user signs back in within 
+10-minutes. The RP should call navigator.credentials.preventSilentAccess() to
```
Comment (agektmr) [inline site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md:653]:
```suggestion
10-minutes. The RP should call `navigator.credentials.preventSilentAccess()` to 
```

**Review by agektmr — APPROVED** [review]
LGTM % comments

File: site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md (hunk 2 above)
Comment (agektmr) [reply site/en/docs/privacy-sandbox/fedcm-developer-guide/index.md:582]:
```suggestion
```

**Review by anusmitaray — APPROVED** [review]

**Pull request merged**